lib/rules/declaration-block-trailing-semicolon/__tests__/index.js

```diff
@@ -25,6 +25,10 @@ testRule(rule, {
 			code: 'a { color: red; &:hover { color: pink; }}',
 			description: 'nesting with first-level decl',
 		},
+		{
```

Comment:
Let's replace these two tests with a single one at the bottom of the file that tests the problematic syntax, e.g.:

```js
testRule(rule, {
	ruleName,
	config: ['always'],
	syntax: 'css-in-js',
	fix: true,

	accept: [
		{
			code: 'const C = () => { return <a style={{ color: "red" }}></a> }',
			description: 'css-in-js object',
		},
	],

	reject: [
		{
			code: 'const C = styled.a`color: red`;',
			fixed: 'const C = styled.a`color: red;`;',
			description: 'css-in-js template literal',
			message: messages.expected,
			line: 1,
			column: 29,
		},
	],
});
```

So that [we're consistent with the conventions](https://stylelint.io/developer-guide/rules#write-tests):

> You should:
> - use standard CSS syntax by default, and only swap parsers when testing a specific piece of non-standard syntax

I think this will also highlight that template literals are incorrectly excluded by the proposed fix.

Reply:
I've updated the snippet above to use the new `testRule` signature because we now dogfood our Jest preset for testing rules (https://github.com/stylelint/stylelint/pull/4758).

Reply:
Ah, very nice. Thanks much!